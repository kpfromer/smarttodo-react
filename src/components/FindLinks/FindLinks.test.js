import React from 'react';
import { shallow } from 'enzyme';

describe('FindLinks', () => {
  let FindLinks, Linkify, Tlds, mockTlds, mockMatch;
  beforeEach(() => {
    jest.doMock('linkify-it');
    jest.doMock('tlds');
    Linkify = require('linkify-it');
    Tlds = require('tlds');

    mockTlds = jest.fn();
    mockMatch = jest.fn();
    
    Linkify.mockReturnValue({
      tlds: mockTlds,
      match: mockMatch
    })

    FindLinks = require('./FindLinks').default;
  });
  afterEach(() => {
    jest.resetAllMocks();
  });
  it('constructs a new linkify with valid tlds', () => {
    shallow(
      <FindLinks handleLink={() => {}}>
        hello
      </FindLinks>
    );
    expect(Linkify).toHaveBeenCalled();
    expect(mockTlds).toHaveBeenCalledWith(Tlds);
  });
  // Finds and replaces with method
  // Check for literal edge cases
  it('returns an array of custom links with text', () => {
    const text = 'http://google.com hello world I need to check out example.com';
    mockMatch
      .mockReturnValue([{
              index: 0,
              lastIndex: 17,
              linkVal: 'google'
            },
            {
              index: 50,
              lastIndex: 62,
              linkVal: 'example'
            }
          ]);
    const handleLink = link => <h1 key={link.linkVal}>{link.linkVal}</h1>;
    const result = shallow(
      <FindLinks handleLink={handleLink}>
        {text}
      </FindLinks>
    );
    expect(result).toMatchSnapshot();
  });
});