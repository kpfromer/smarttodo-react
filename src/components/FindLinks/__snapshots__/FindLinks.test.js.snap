// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindLinks returns an array of custom links with text 1`] = `
<Fragment>
  <h1
    key="google"
  >
    google
  </h1>
   hello world I need to check out 
  <h1
    key="example"
  >
    example
  </h1>
</Fragment>
`;
